snowflake/JsonUploadFromBlob.txt

```diff
@@ -0,0 +1,123 @@
+-- SELECT DATABASE --
+use database BLOB_DB;
+
+use schema PUBLIC;
+
+use role accountadmin;
+
+-- Create stage --
+
+create or replace stage azure_stage_sas
+  URL='azure://<storage_account>.blob.core.windows.net/<container>'
+  CREDENTIALS=(AZURE_SAS_TOKEN='?<sas-token>');
+
+list @azure_stage_sas;
+
+-- Import raw data to table --
+
+create or replace table RAW_SOURCE (
+  SRC VARIANT);
+
+create or replace task copy_data
+warehouse=PLAYWH
+schedule='1 MINUTE'
+AS
+copy into RAW_SOURCE from @azure_stage_sas
+FILE_FORMAT=(TYPE=json);
+
+alter task copy_data resume;
+
+-- Create source table with hash--
+
+create or replace table RAW_SOURCE_HASHED as select * from RAW_SOURCE;
+
+alter table RAW_SOURCE_HASHED
+add v_sha varchar;
+
+update RAW_SOURCE_HASHED SET
+v_sha=SHA2(SRC);
+
+-- Insert data to table with hash --
+
+create or replace task merge_sources
+warehouse=PLAYWH
+schedule='1 MINUTE'
+AS
+merge into RAW_SOURCE_HASHED using RAW_SOURCE
+on RAW_SOURCE_HASHED.SRC=RAW_SOURCE.SRC
+when matched then update set RAW_SOURCE_HASHED.SRC=RAW_SOURCE.SRC
+when not matched then insert (SRC) values (RAW_SOURCE.SRC);
+
+alter task merge_sources resume;
+
+create or replace task generate_sha
+warehouse=PLAYWH
+schedule='1 MINUTE'
+AS
+update RAW_SOURCE_HASHED SET
+v_sha=SHA2(SRC)
+;
+
+alter task generate_sha resume;
+
+-- Create users table --
+
+create or replace table USERS_TRANSFORMED as

```

Comment:
It's a really nice way to create a table. However, in this case we need explicit columns declaration in order to set validation constraints (e.g. VARCHAR(30), etc.)

Reply:
Or, if it's possible to set this constraints with this notation (e.g. `src:FullName::VARCHAR(30)` instead of `src:FullName::string`) let's do it

Reply:
Changed table creation method